src/mapeo-project.js

```diff
@@ -876,25 +897,26 @@ async function deleteAll(dataType) {
  * @param {MapeoProject['$translation']['dataType']} opts.translation
  * @param {MapeoProject['preset']} opts.preset
  * @param {MapeoProject['field']} opts.field
+ * @returns {Promise<String[]>}
  */
-async function deleteTranslations(opts) {
+async function grabTranslationsToDelete(opts) {
+  /** @type {String[]} */
+  const toDelete = []
   const translations = await opts.translation.getMany()
-  await Promise.all(
-    translations.map(async ({ docId, docRef, docRefType }) => {
-      if (docRefType === 'preset' || docRefType === 'field') {
-        let shouldDelete = false
-        try {
-          const toDelete = await opts[docRefType].getByDocId(docRef.docId)
-          shouldDelete = toDelete.deleted
-        } catch (e) {
-          opts.logger.log(`referred ${docRef.docId} is not found`)
-        }
-        if (shouldDelete) {
-          await opts.translation.delete(docId)
-        }
+  translations.forEach(async ({ docRefType, docRef, docId }) => {
```

Comment:
I think we need to `await` this.

Reply:
doesn't seem to change anything, and the linter complains that 'await has no effect on this type of expression' ...